### DISCUSSION TITLE
LDAP protocol in Linux Docker

### DISCUSSION THREAD

**Discussion by istretyakov** [dotnet/runtime#51511] [Q&A]
Hi!

I try to send LDAP request from Docker container using System.DirectoryServices.Protocols 5.0.0 on .NET 5.0.
In Windows-case i get response that server is unavailable. It's right. This response shows that request was sent. In other environment (work) i get users. With Docker and without Docker, on Windows.
But in Linux-case (Docker) each time i have this exception: The LDAP server returned an unknown error. ErrorCode -1. I can write not existed server and i will get this exception. It means that doesn't work. And i added installing libldap-2.4-2 in Dockerfile.
One code, different reactions.

Windows:
![Windows2](https://user-images.githubusercontent.com/25767039/115268126-9a165380-a142-11eb-85de-1b563c5b7648.jpg)

Linux:
![Linux2](https://user-images.githubusercontent.com/25767039/115268128-9b478080-a142-11eb-96d2-e2706fc30c58.jpg)

**Comment by bc4250** [comment]
try this

AuthType = System.DirectoryServices.Protocols.AuthType.Basic

**Comment by SommerEngineering** [comment]
We also had many problems with LDAP on Linux using `System.DirectoryServices.Protocols.*`. In the end, only one thing helped: The migration to [Novell.Directory.Ldap.NETStandard](https://github.com/dsbenghe/Novell.Directory.Ldap.NETStandard). This allowed us to reliably use LDAP under Linux 👍 .
